pubsub-mapped-api/src/main/java/com/google/pubsub/clients/consumer/KafkaConsumer.java

```diff
@@ -0,0 +1,592 @@
+import com.google.pubsub.kafkastubs.common.serialization.Deserializer;
+import com.google.pubsub.kafkastubs.consumer.Consumer;
+import com.google.pubsub.kafkastubs.consumer.ConsumerRebalanceListener;
+import com.google.pubsub.kafkastubs.consumer.ConsumerRecord;
+import com.google.pubsub.kafkastubs.consumer.ConsumerRecords;
+import com.google.pubsub.kafkastubs.consumer.OffsetAndMetadata;
+import com.google.pubsub.kafkastubs.consumer.OffsetAndTimestamp;
+import com.google.pubsub.kafkastubs.consumer.OffsetCommitCallback;
+import com.google.pubsub.kafkastubs.utils.Utils;
+import com.google.pubsub.v1.AcknowledgeRequest;
+import com.google.pubsub.v1.DeleteSubscriptionRequest;
+import com.google.pubsub.v1.ListTopicsRequest;
+import com.google.pubsub.v1.ListTopicsResponse;
+import com.google.pubsub.v1.PublisherGrpc.PublisherBlockingStub;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.PullRequest;
+import com.google.pubsub.v1.PullResponse;
+import com.google.pubsub.v1.ReceivedMessage;
+import com.google.pubsub.v1.SubscriberGrpc.SubscriberBlockingStub;
+import com.google.pubsub.v1.SubscriberGrpc.SubscriberFutureStub;
+import com.google.pubsub.v1.Subscription;
+import com.google.pubsub.v1.Topic;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+import org.apache.commons.lang3.NotImplementedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KafkaConsumer<K, V> implements Consumer<K, V> {
+
+  private static final Logger log = LoggerFactory.getLogger(KafkaConsumer.class);
+  private static final String GOOGLE_CLOUD_PROJECT;
+  private static final String SUBSCRIPTION_PREFIX;
+  private static final String TOPIC_PREFIX;
+
+  //because of no partition concept, the partition number is constant
+  private static final int DEFAULT_PARTITION = 0;
+
+  //because checksum is not needed, it is defined as a constant
+  private static final int DEFAULT_CHECKSUM = 1;
+
+  private static final String KEY_ATTRIBUTE = "key_attribute";
+
+  private StubCreator stubCreator;
+  private ImmutableMap<String, Subscription> subscriptions = ImmutableMap.of();
+
+  private final Deserializer<K> keyDeserializer;
+  private final Deserializer<V> valueDeserializer;
+
+  private final int maxPollRecords;
+
+  static {
+    GOOGLE_CLOUD_PROJECT = "projects/" + System.getenv("GOOGLE_CLOUD_PROJECT");
+    SUBSCRIPTION_PREFIX = GOOGLE_CLOUD_PROJECT + "/subscriptions/";
+    TOPIC_PREFIX = GOOGLE_CLOUD_PROJECT + "/topics/";
+  }
+
+  public KafkaConsumer(Map<String, Object> configs) {
```

Comment:
KafkaConsumer done by Kafka is providing 4 public constructors. We decided that we want to allow users to switch as smooth as we can, so i don't think we cal delete any of these.